libraries/apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/introspection/IntrospectionSchema.kt

```diff
@@ -144,9 +157,66 @@ data class IntrospectionSchema(
         val ofType: TypeRef? = null,
     )
 
+    /**
+     * An introspection directive
+     */
+    @Serializable
+    data class Directive(
+        val name: String,
+        val description: String?,
+        val locations: List<DirectiveLocation> = emptyList(),
+        val args: List<Argument>,
+        val isRepeatable: Boolean = false,
+    ) {
+      @Serializable
+      data class Argument(
+          val name: String,
+          val description: String?,
+          val isDeprecated: Boolean = false,
+          val deprecationReason: String?,
+          val type: TypeRef,
+          val defaultValue: String?,
+      )
+
+      enum class DirectiveLocation {
+        QUERY,
+        MUTATION,
+        SUBSCRIPTION,
+        FIELD,
+        FRAGMENT_DEFINITION,
+        FRAGMENT_SPREAD,
+        INLINE_FRAGMENT,
+        VARIABLE_DEFINITION,
+        SCHEMA,
+        SCALAR,
+        OBJECT,
+        FIELD_DEFINITION,
+        ARGUMENT_DEFINITION,
+        INTERFACE,
+        UNION,
+        ENUM,
+        ENUM_VALUE,
+        INPUT_OBJECT,
+        INPUT_FIELD_DEFINITION,
+      }
+    }
+
     enum class Kind {
       ENUM, INTERFACE, OBJECT, INPUT_OBJECT, SCALAR, NON_NULL, LIST, UNION
     }
+
+    fun copy(
+        queryType: QueryType = this.queryType,
+        mutationType: MutationType? = this.mutationType,
+        subscriptionType: SubscriptionType? = this.subscriptionType,
+        types: List<Type> = this.types,
+    ) = copy(
+        queryType = queryType,
+        mutationType = mutationType,
+        subscriptionType = subscriptionType,
+        types = types,
+        directives = emptyList(),
+    )
```

Comment:
I'm guessing this is a binary compatibility method? If yes, the problem is that it shadows the default `.copy` and gets resolved from `normalize()` for an example. 

Maybe make it hidden? 

```suggestion
    @Deprecated("For binary compatibility only.", level = DeprecationLevel.HIDDEN)
    fun copy(
        queryType: QueryType = this.queryType,
        mutationType: MutationType? = this.mutationType,
        subscriptionType: SubscriptionType? = this.subscriptionType,
        types: List<Type> = this.types,
    ) = copy(
        queryType = queryType,
        mutationType = mutationType,
        subscriptionType = subscriptionType,
        types = types,
        directives = emptyList(),
    )
```

Reply:
Exactly - good catch! 👍

Reply:
Mmm looks like that failed in `apiCheck` because the method is marked as 'synthetic'. I _think_ this is ok though cause it's still binary compatible.